addons/portal/controllers/portal.py

```diff
@@ -31,20 +36,39 @@ def pager(url, total, page=1, step=30, scope=5, url_args=None):
     page_count = int(math.ceil(float(total) / step))
 
     page = max(1, min(int(page if str(page).isdigit() else 1), page_count))
-    scope -= 1
-
-    pmin = max(page - int(math.floor(scope/2)), 1)
-    pmax = min(pmin + scope, page_count)
 
-    if pmax - pmin < scope:
-        pmin = pmax - scope if pmax - scope > 0 else 1
+    page_previous = max(1, page - 1)
+    page_next = min(page_count, page + 1)
 
     def get_url(page):
         _url = "%s/page/%s" % (url, page) if page > 1 else url
         if url_args:
             _url = "%s?%s" % (_url, urls.url_encode(url_args))
         return _url
 
+    def add_pages(page_list):
+        """ Helper to add a page or ellipsis to the pages list. """
+        for current_page in page_list:
+            pages.append({
+                'num': current_page,
+                'url': get_url(current_page) if current_page != '...' else None,
+                'is_current': current_page == page
+            })
+
+    pages = []
+
+    if page_count <= 5:
+        # Show all pages if there are 5 or fewer pages
+        add_pages(range(1, page_count + 1))
+    else:
+        if page <= 3:
+            add_pages([1, 2, 3, 4, '...', page_count])
+        elif page >= page_count - 2:
+            add_pages([1, '...', page_count - 3, page_count - 2, page_count - 1, page_count])
+        else:
+            # For middle pages, show ellipses before and after the current page
+            add_pages([1, '...', page - 1, page, page + 1, '...', page_count])
```

Comment:
Swap elif else

elfi pag < pag_count - 2 and then else
to increase readability 
if 1 to 5
elif 6 to x - 2
else x -2 to x

Reply:
Yes, seems good!